From 672f0d79b50b6cd976c706197be34e5a7a7a01af Mon Sep 17 00:00:00 2001
From: Bard Liao <bardliao@realtek.com>
Date: Thu, 5 Feb 2015 16:41:18 +0530
Subject: [PATCH] ASoC: rt5670: Turn On the PLL during Speaker PB

The Speaker power is turned on during Speaker PB

Issue: GMINL-6938

Change-Id: Ie02bc929457eed8c2229e48e3246fdaa811efdaf
Signed-off-by: Bard Liao <bardliao@realtek.com>
Signed-off-by: Samreen Nilofer <samreen.nilofer@intel.com>
---
 sound/soc/codecs/rt5670.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/soc/codecs/rt5670.c b/sound/soc/codecs/rt5670.c
index 2516871..5bb1535 100644
--- a/sound/soc/codecs/rt5670.c
+++ b/sound/soc/codecs/rt5670.c
@@ -2431,6 +2431,10 @@ static const struct snd_soc_dapm_route rt5670_dapm_routes[] = {
 	{ "micbias1", NULL, "DAC L1 Power" },
 	{ "micbias1", NULL, "DAC R1 Power" },
 
+	{ "dac stereo1 filter", NULL, "PLL1", check_sysclk1_source },
+	{ "dac mono left filter", NULL, "PLL1", check_sysclk1_source },
+	{ "dac mono right filter", NULL, "PLL1", check_sysclk1_source },
+
 	{ "DMIC1", NULL, "DMIC L1" },
 	{ "DMIC1", NULL, "DMIC R1" },
 	{ "DMIC2", NULL, "DMIC L2" },
@@ -2723,14 +2727,10 @@ static const struct snd_soc_dapm_route rt5670_dapm_routes[] = {
 
 	{ "DAC L1", NULL, "DAC L1 Power" },
 	{ "DAC L1", NULL, "Stereo DAC MIXL" },
-	{ "DAC L1", NULL, "PLL1", check_sysclk1_source },
 	{ "DAC R1", NULL, "DAC R1 Power" },
 	{ "DAC R1", NULL, "Stereo DAC MIXR" },
-	{ "DAC R1", NULL, "PLL1", check_sysclk1_source },
 	{ "DAC L2", NULL, "Mono DAC MIXL" },
-	{ "DAC L2", NULL, "PLL1", check_sysclk1_source },
 	{ "DAC R2", NULL, "Mono DAC MIXR" },
-	{ "DAC R2", NULL, "PLL1", check_sysclk1_source },
 
 	{ "OUT MIXL", "BST1 Switch", "BST1" },
 	{ "OUT MIXL", "INL Switch", "INL VOL" },
-- 
1.9.1

